test_wikipron.py

```diff
@@ -200,6 +201,46 @@ def test_li_selector(dialect, require_dialect_label, expected_li_selector):
     assert config.li_selector == expected_li_selector
 
 
+def test_american_english_dialect_selection():
+    # Pick a word for which Wiktionary has dialect-specified pronunciations
+    # for both US and non-US English.
+    word = "mocha"
+    html_session = requests_html.HTMLSession()
+    response = html_session.get(_PAGE_TEMPLATE.format(word=word))
+    # Construct two configs to demonstrate the US dialect (non-)selection.
+    config_only_us = _config_factory(key="en", dialect="US | American English")
+    config_any_dialect = _config_factory(key="en")
+    # Apply each config's XPath selector.
+    results_only_us = response.html.xpath(config_only_us.li_selector)
+    results_any_dialect = response.html.xpath(config_any_dialect.li_selector)
+    assert (
+        len(results_any_dialect)  # containing both US and non-US results
+        > len(results_only_us)  # containing only the US result
+        > 0
+    )
+
+
+def test_require_dialect_label():
+    # Pick a word for which Wiktionary doesn't specify the dialect at all.
+    word = "examine"
+    html_session = requests_html.HTMLSession()
+    response = html_session.get(_PAGE_TEMPLATE.format(word=word))
+    # Construct two configs to test the "require_dialect_label" param.
+    config_params = dict(key="en", dialect="US | American English")
+    config_dialect_optional = _config_factory(**config_params)
+    config_dialect_required = _config_factory(
+        **config_params, require_dialect_label=True
+    )
+    # Apply each config's XPath selector.
+    results_dialect_optional = response.html.xpath(
+        config_dialect_optional.li_selector
+    )
+    results_dialect_required = response.html.xpath(
+        config_dialect_required.li_selector
+    )
+    assert len(results_dialect_optional) > len(results_dialect_required) == 0
```

Comment:
`test_american_english_dialect_selection` and `test_require_dialect_label` might look pretty similar for their approach, but I've decided to keep them apart as opposed to trying to merge them as one test, for better readability.

Reply:
These two tests require the Internet connection, which I feel fine with. Let me know if you feel strongly about adding `pytest.mark.skipif` to avoid confusion when we run the test suite locally without Internet somehow.

Reply:
What's the `skipif` look like for testing for a WAN connection? I think it wouldn't be crazy---I do a lot of hobbyist development on planes and buses.

Reply:
The `skipif` is a decorator on top of a test that you'd like to skip if a given condition is true. It sounds you might want the connectivity check -- I've added it at c965a26. (I didn't spend too much time thinking about how to implement a WAN connectivity check, how to optimize it, etc.)